pkg/api/entries.go

```diff
@@ -251,25 +263,44 @@ func SearchLogQueryHandler(params entries.SearchLogQueryParams) middleware.Respo
 	}
 
 	if len(params.Entry.LogIndexes) > 0 {
-		leaves := []*trillian.LogLeaf{}
-		for _, logIndex := range params.Entry.LogIndexes {
-			resp := tc.getLeafByIndex(swag.Int64Value(logIndex))
-			switch resp.status {
-			case codes.OK, codes.NotFound:
-			default:
-				return handleRekorAPIError(params, http.StatusInternalServerError, fmt.Errorf("grpc error: %w", resp.err), trillianUnexpectedResult)
+		g, _ := errgroup.WithContext(httpReqCtx)
+
+		leaves := make([]*trillian.LogLeaf, len(params.Entry.LogIndexes))
+		for i, logIndex := range params.Entry.LogIndexes {
+			i, logIndex := i, logIndex // https://golang.org/doc/faq#closures_and_goroutines
+			g.Go(func() error {
+				resp := tc.getLeafByIndex(swag.Int64Value(logIndex))
```

Comment:
The trillian client itself (not our TrillianClient, but tc.Client) has a GetLeavesByIndex that takes a slice of indices. Do you think that one would be better here?

Reply:
I had looked at that, but discovered that https://github.com/google/trillian/pull/2244 (which has already merged)... so since it appears to be deprecated/removed in the next release, I steered away from it. 

The `GetLeavesByRange` call could be used to optimize this a bit, but I didn't think it was worth doing right now.

Reply:
Ah, good catch!